### PR TITLE
Pass subset of ClimaAtmosParameters to detrainment

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR passes a subset of ClimaAtmosParameters to detrainment, as we are near hitting the limit of parameter memory space when running on the gpu.

### CODE REVIEW

**Review by szy21 — APPROVED** [review]
Thanks! We can just replace `turbconv_params` with `tc_params` in some of the functions?

File: src/prognostic_equations/edmfx_entr_detr.jl [hunk 1]
```diff
@@ -100,8 +106,8 @@ function entrainment(
     ᶜtke⁰,
     ::GeneralizedEntrainment,
 )
-    FT = eltype(params)
-    turbconv_params = CAP.turbconv_params(params)
+    FT = eltype(thermo_params)
+    turbconv_params = turbconv_params
```
Comment (szy21) [inline src/prognostic_equations/edmfx_entr_detr.jl:110]:
```suggestion
```

File: src/prognostic_equations/edmfx_entr_detr.jl [hunk 2]
```diff
@@ -271,8 +282,8 @@ function detrainment(
     ᶜtke⁰,
     ::GeneralizedDetrainment,
 )
-    FT = eltype(params)
-    turbconv_params = CAP.turbconv_params(params)
+    FT = eltype(thermo_params)
+    turbconv_params = turbconv_params
```
Comment (szy21) [inline src/prognostic_equations/edmfx_entr_detr.jl:286]:
```suggestion
```

**Comment by szy21** [comment]
Hmm, somehow the reproducibility test failed. I don't think this should change MSE though?

File: src/prognostic_equations/edmfx_entr_detr.jl (hunk 1 above)
Comment (charleskawczynski) [reply src/prognostic_equations/edmfx_entr_detr.jl]: Good catch, thanks!

**Comment by charleskawczynski** [comment]
> Hmm, somehow the reproducibility test failed. I don't think this should change MSE though?

Yeah, that's strange. My best guess is that it ran on a different machine? Or there are still issues with the reproducibility test infrastructure that need worked out. I'm working on adding more unit tests now.

**Comment by charleskawczynski** [comment]
Please let me know if this was sufficient to get your example working (cc @szy21). If not, I can do a followup with updating the actual broadcast expression, which I think could have a bigger impact since the local geometry is passed in.

**Pull request merged**